src/txn/txn.c

```diff
@@ -1669,13 +1669,38 @@ __txn_mod_compare(const void *a, const void *b)
     return (0);
 }
 
+/*
+ * __txn_check_if_stable_has_moved_ahead_commit_ts --
+ *     Check if the stable timestamp has moved ahead of the commit timestamp.
+ */
+static int
+__txn_check_if_stable_has_moved_ahead_commit_ts(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_TXN *txn;
+    WT_TXN_GLOBAL *txn_global;
+
+    conn = S2C(session);
+    txn = session->txn;
+    txn_global = &conn->txn_global;
+
+    if (txn_global->has_stable_timestamp && txn->first_commit_timestamp != 0 &&
+      txn_global->stable_timestamp >= txn->first_commit_timestamp)
+        WT_RET_MSG(session, WT_ROLLBACK,
```

Comment:
I think we should also update the rollback reason field.

Reply:
I think we probably should return EINVAL instead of WT_ROLLBACK

Reply:
Fixed in f08e9354669258aef7f1ed7607f4960fabd9ae1b